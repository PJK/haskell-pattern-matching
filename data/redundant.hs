module Redundant where

-- Figure out a way to get the base data constructors
-- We may have to rename all our internal representations of names because of possible overlaps,
-- but I really don't want to have to go and look through import lists.
data MyBool
    = True
    | False

func :: Bool -> Int
func True = 1
func False = 2
func _ = 3
