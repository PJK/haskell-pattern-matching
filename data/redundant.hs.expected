[
    [
        [
            [
                {
                    "tag": "ConstructorPattern",
                    "contents": [
                        "True",
                        []
                    ]
                }
            ],
            [
                {
                    "tag": "ConstructorPattern",
                    "contents": [
                        "False",
                        []
                    ]
                }
            ]
        ],
        [
            [
                [
                    [
                        {
                            "tag": "WildcardPattern",
                            "contents": []
                        }
                    ],
                    [
                        {
                            "tag": "EvaluatedConstructor",
                            "contents": [
                                "True",
                                []
                            ]
                        },
                        {
                            "tag": "EvaluatedConstructor",
                            "contents": [
                                "False",
                                []
                            ]
                        }
                    ]
                ]
            ]
        ]
    ]
]